src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -328,7 +331,7 @@ public async void WorkerError(WorkerErrorEvent workerError)
 
         public async void WorkerRestart(WorkerRestartEvent workerRestart)
         {
-            if (!_disposing)
+            if (!_disposing && !_disposed)
```

Comment:
same here. change to 
```
if (_disposing || _disposed)
            {
                return;
            }
```

Reply:
Updated the pattern here and in other places too